src/DataFixtures/Faker/Provider/UlidProvider.php

```diff
@@ -2,13 +2,89 @@
 
 namespace App\DataFixtures\Faker\Provider;
 
+use Faker\Factory;
 use Faker\Provider\Base;
+use Symfony\Component\Uid\BinaryUtil;
 use Symfony\Component\Uid\Ulid;
 
 class UlidProvider extends Base
 {
-    public static function ulid()
+    private const NIL = '00000000000000000000000000';
+
+    private static string $time = '';
+    private static array $rand = [];
+
+    public static function ulid(): Ulid
+    {
+        $ulid = self::doGenerate();
+
+        return new Ulid($ulid);
+    }
+
+    public static function ulidDate(Ulid $ulid): \DateTime
+    {
+        return \DateTime::createFromImmutable($ulid->getDateTime());
+    }
+
+    private static function doGenerate(string $mtime = null): string
     {
-        return new Ulid();
+        $faker = Factory::create();
+
+        if (null === $time = $mtime) {
+            // $time = microtime(false);
```

Comment:
Remove commented out code.
Maybe add a description in the top of the function about where this is from and what changes have been made.